driver/pparse.ml

```diff
@@ -79,9 +85,10 @@ let apply_rewriter magic fn_in ppx =
   end;
   fn_out
 
-let read_ast magic fn =
+let read_ast (type a) (kind : a ast_kind) fn : a =
   let ic = open_in_bin fn in
   try
+    let magic = magic_of_kind kind in
     let buffer = really_input_string ic (String.length magic) in
     assert(buffer = magic); (* already checked by apply_rewriter *)
```

Comment:
I tried to do this, but it's a bit of a hassle because I'm not sure how to raise the error. Should I `raise (Error (WrongMagic ...))` as in `apply_rewriter` above? But currently WrongMagic expects a "command" and its error message is tailored for that; so maybe add another constructor? Or have `WrongMagic` take a `(Command of ... | File of ...)` parameter -- breaking compatibility? What would you do?